eos_data_distribution/ndn/command.py

```diff
@@ -0,0 +1,101 @@
+# -*- Mode:python; coding: utf-8; c-file-style:"gnu"; indent-tabs-mode:nil -*- */
+#
+# Copyright (C) 2016 Endless Mobile INC.
+# Author: Niv Sardi <xaiki@endlessm.com>
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+# A copy of the GNU Lesser General Public License is in the file COPYING.
+
+import logging
+
+from pyndn.control_parameters import ControlParameters
+from pyndn.encoding.tlv_wire_format import TlvWireFormat
+from pyndn.util.command_interest_generator import CommandInterestGenerator
+from pyndn import Interest, Name
+
+logger = logging.getLogger(__name__)
+
+_commandInterestGenerator = CommandInterestGenerator()
+
+def makeCommandInterest (*args, **kwargs):
+    logger.debug("args kwargs: %s %s", args, kwargs)
+    _commandInterestGenerator.generate(*args, **kwargs)
+
+controlMap = {
+    'name': 'setName',
+    'faceid': 'setFaceId',
+    'uri' : 'setUri',
+    'control': 'setLocalControlFeature',
+    'origin': 'setOrigin',
+    'cost': 'setCost',
+    'flags': 'setFlags',
+    'mask': 'setMask',
+    'strategy': 'setStrategy',
+    'expiration': 'setExpirationPeriod',
+    'persistency': 'setFacePersistency',
+    'flagbit': 'setFlagBit'
+}
+
+def makeInterest(cmd, flags=None, local=True,
+                 keyChain=None, certificateName=None,
+                 controlParameters={}):
+    cp = ControlParameters()
+
+    logger.debug('control Parameters: %s', controlParameters)
+    for c in controlParameters:
+        getattr(cp,controlMap[c])(controlParameters[c])
```

Comment:
Sorry but I really don't like this kind of generic coding. Just have the creator pass in a `ControlParameters` instance.